backend/src/main/java/vn/edu/hust/nmcnpm_20242_n3/service/BookRequestService.java

```diff
@@ -0,0 +1,66 @@
+package vn.edu.hust.nmcnpm_20242_n3.service;
+
+import jakarta.transaction.Transactional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import vn.edu.hust.nmcnpm_20242_n3.entity.BookLoan;
+import vn.edu.hust.nmcnpm_20242_n3.entity.BookRequest;
+import vn.edu.hust.nmcnpm_20242_n3.repository.BookRequestRepository;
+import vn.edu.hust.nmcnpm_20242_n3.constant.BookLoanStatusEnum;
+import vn.edu.hust.nmcnpm_20242_n3.constant.BookRequestStatusEnum;
+import vn.edu.hust.nmcnpm_20242_n3.constant.BookRequestTypeEnum;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+@Service
+public class BookRequestService {
+
+    private final BookRequestRepository bookRequestRepository;
+    private final BookLoanService bookLoanService;
+
+    @Autowired
+    public BookRequestService(BookRequestRepository bookRequestRepository, BookLoanService bookLoanService) {
+        this.bookRequestRepository = bookRequestRepository;
+        this.bookLoanService = bookLoanService;
+    }
+
+    public List<BookRequest> getAllPendingRequests() {
+        return bookRequestRepository.findByStatusOrderByCreatedAtAsc(BookRequestStatusEnum.PENDING);
+    }
+
+    @Transactional
+    public BookRequest processRequest(Integer requestId, boolean approve) {
+        BookRequest request = bookRequestRepository.findById(requestId)
+                .orElseThrow(() -> new IllegalArgumentException("Request not found with ID: " + requestId));
```

Comment:
I think we can only approve/reject a `PENDING` request. Should we change the conditions here a bit?